File: BOJ/10XXX/10773-제로.js
/**
* ABOUT
*
* DATE: 2021-05-23
* AUTHOR: lovelysi0113
*
* URL: https://www.acmicpc.net/problem/10773
*
* COMMENT:
* 현재 수가 숫자면 stack에 넣고(push), 0이면 stack에서 뺀다(pop)
* 모든 작업이 끝난 후, stack에 남아있는 모든 수의 합을 구하면 된다
**/

const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let input = [];

rl.on('line', function (line) {
    input.push(parseInt(line));
})
.on('close', function () {
    let account = [];
    input.slice(1).forEach(num => {
        if (num === 0) {
            account.pop();
        } else {
            account.push(num);
        }
    });
    
    let answer = account.reduce((sum, num) => sum + num, 0);
    console.log(answer);
    
    process.exit();
});

File: BOJ/10XXX/10773-제로.py
'''/**
* ABOUT
*
* DATE: 2021-05-23
* AUTHOR: lovelysi0113
*
* URL: https://www.acmicpc.net/problem/10773
*
* COMMENT:
* 현재 수가 숫자면 stack에 넣고(push), 0이면 stack에서 뺀다(pop)
* 모든 작업이 끝난 후, stack에 남아있는 모든 수의 합을 구하면 된다
**/'''

cnt = int(input())

account = []
for _ in range(cnt):
    num = int(input())
    if num == 0:
        account.pop()
    else:
        account.append(num)

print(sum(account))
